### PR TITLE
Update to DSL2 Best Practices

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by maxulysse** [comment]
Following the @drpatelh DSL2 Best Pratices

**Comment by maxulysse** [comment]
OK, so now back on track, with some tiny added improvements:
With https://github.com/nf-core/test-datasets/pull/262, I introduced samplesheet.csv files in sarek
With https://github.com/nf-core/test-datasets/pull/263, I want to try out how minimal such a file can be within sarek, and how flexible we can be with our csv files.

I really love the samplesheet check process introduced by @drpatelh in the DSL2 pipeline, but I feel that we're still wasting a process for that, which is what I'm exploring here in Sarek:
https://github.com/nf-core/sarek/blob/4fbbf80a90e4557b069435c86ca95bebd66e5b78/workflows/sarek.nf#L376-L412

Plan is to asses as much as we can from the header and infer the rest.
So won't change much for our regular users (apart from switching to DSL2 and having tons of images instead of one, and switching from a tsv to a csv file with a header).
But it could simplify access for new users and we don't need all the information all the time.
We only need gender when doing cnv, and we only need status for tumor/normal pairs.
So if it's not specified, why not assigning by default a 0 as status and "NA" as gender?
Can we remove anything else?
Do we need any more information?

@nf-core/core  @nf-core/sarek what do you think?

**Comment by maxulysse** [comment]
I'll fix CI when I'll update modules

**Comment by maxulysse** [comment]
ok, now that I fixed CI once again, I'll now update all current modules

File: conf/test.config [hunk 1]
```diff
@@ -31,6 +31,61 @@ params {
   schema_ignore_params = 'genomes,input'
 }
 
+profiles {
```
Comment (FriederikeHanssen) [inline conf/test.config:34]: ah nice, this gives a nice overview :)

File: main.nf [hunk 2]
```diff
@@ -1,97 +1,87 @@
...
-}
+/*
+========================================================================================
+    GENOME PARAMETER VALUES
+========================================================================================
+*/
 
-// Check AWS batch settings
-Checks.aws_batch(workflow, params)
+params.ac_loci                 = WorkflowMain.getGenomeAttribute(params, 'ac_loci')
```
Comment (FriederikeHanssen) [inline main.nf:37]: love it

File: modules/nf-core/software/samtools/stats/main.nf [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
 process SAMTOOLS_STATS {
     tag "$meta.id"
+    label 'process_low'
     publishDir "${params.outdir}",
         mode: params.publish_dir_mode,
-        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
 
-    conda (params.enable_conda ? "bioconda::samtools=1.12" : null)
+    conda (params.enable_conda ? "bioconda::samtools=1.10" : null)
```
Comment (FriederikeHanssen) [inline modules/nf-core/software/samtools/stats/main.nf:14]: why did you downgrade samtools here?

**Review by FriederikeHanssen — APPROVED** [review]
looks good. This is the PR depending on something weird with the test data, right?

File: modules/nf-core/software/samtools/stats/main.nf (hunk 3 above)
Comment (maxulysse) [reply modules/nf-core/software/samtools/stats/main.nf:14]: mistake, I'll fix that in the next PR.
Thanks for spotting that

File: conf/test.config (hunk 1 above)
Comment (maxulysse) [reply conf/test.config:34]: Yes, I wanted to simplify all that

**Comment by maxulysse** [comment]
> looks good. This is the PR depending on something weird with the test data, right?

Just with `CSV` instead of `TSV` actually.
(but yeah now we can go weird with the `CSV` too).

File: modules/nf-core/software/samtools/stats/main.nf (hunk 3 above)
Comment (maxulysse) [reply modules/nf-core/software/samtools/stats/main.nf:14]: cf https://github.com/nf-core/sarek/pull/386

**Pull request merged**